utils/collectd/files/exec-scripts/dslstat.lua

```diff
@@ -0,0 +1,150 @@
+-- Copyright 2021 Florian Eckert <fe@dev.tdt.de>
+-- Copyright 2021 Jeroen Peelaerts <jeroen.peelaerts@gmail.com>
+-- Licensed to the public under the GNU General Public License v2.
+
+local ubus = require("ubus")
+local fs = require("nixio.fs")
+
+local hostname_file = "/proc/sys/kernel/hostname"
+
+local line_vars = {
+	{
+		name = "vector",
+		type = "bool",
+	},
+	{
+		name = "trellis",
+		type = "bool"
+	},
+	{
+		name = "bitswap",
+		type = "bool"
+	},
+	{	
+		name = "retx",
+		type = "bool",
+	},
+	{
+		name = "satn",
+		type = "snr"
+	},
+	{
+		name = "latn",
+		type = "snr"
+	},
+	{
+		name = "attndr",
+		type = "bitrate"
+	},
+	{
+		name = "snr",
+		type = "snr",
+	},
+	{
+		name = "data_rate",
+		type = "bitrate"
+	},
+	{
+		name = "latn",
+		type = "latency"
+	}
+}
+
+local errors = {
+	{
+		name = "ses",
+		type = "gauge"
+	},
+	{
+		name = "uas",
+		type = "gauge"
+	},
+	{
+		name = "rx_corrupted",
+		type = "gauge",
+	},
+	{
+		name = "rx_retransmitted",
+		type = "gauge",
+	},
+	{
+		name = "tx_retransmitted",
+		type = "gauge",
+	}
+}
+
+
+local general_vars = {
+
+	{
+		name = "profile",
+		type = "gauge"
+	},
+	{
+		name = "mode",
+		type = "gauge",
+	},
+	{
+		name = "state",
+		type = "gauge"
+	},
+	{
+		name = "power_state",
```

Comment:
Power state is now a string.

Reply:
I tried to change it to string but then saw some collectd errors. Not sure if this a default type in the `types.db`?